tests/e2e/e2e_permissive_smi_switching.go

```diff
@@ -0,0 +1,290 @@
+						for _, podResult := range ns {
+							if podResult.StatusCode == 200 {
+								return false
+							}
+						}
+					}
+					return true
+				}, 5, 150*time.Second)
+				Expect(success).To(BeTrue())
+
+				By("Re-enable Permissive mode, check again traffic with targets within the mesh")
+				Expect(setPermissiveMode(true)).To(BeNil())
+
+				success = Td.WaitForRepeatedSuccess(func() bool {
+					// Issue all calls, get results
+					results = Td.MultipleHTTPRequest(&requests)
+
+					// Care, depending on variables there could be a lot of results
+					Td.PrettyPrintHTTPResults(&results)
+
+					// Verify success conditions
+					for _, ns := range results {
+						for _, podResult := range ns {
+							if podResult.Err != nil || podResult.StatusCode != 200 {
+								return false
+							}
+						}
+					}
+					return true
+				}, 5, 150*time.Second)
+				Expect(success).To(BeTrue())
+
+				By("Install SMI rules, expect traffic nothing to be affected")
+				for _, srcClient := range sourceNamespaces {
+					httpRG, trafficTarget := Td.CreateSimpleAllowPolicy(
+						SimpleAllowPolicy{
+							RouteGroupName:    srcClient,
+							TrafficTargetName: srcClient,
+
+							SourceNamespace:      srcClient,
+							SourceSVCAccountName: srcClient,
+
+							DestinationNamespace:      destApp,
+							DestinationSvcAccountName: destApp,
+						})
+
+					// Configs have to be put into a monitored NS, and osm-system can't be by cli
+					_, err = Td.CreateHTTPRouteGroup(srcClient, httpRG)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = Td.CreateTrafficTarget(srcClient, trafficTarget)
+					Expect(err).NotTo(HaveOccurred())
+				}
+
+				success = Td.WaitForRepeatedSuccess(func() bool {
+					// Issue all calls, get results
+					results = Td.MultipleHTTPRequest(&requests)
+
+					// Care, depending on variables there could be a lot of results
+					Td.PrettyPrintHTTPResults(&results)
+
+					// Verify success conditions
+					for _, ns := range results {
+						for _, podResult := range ns {
+							if podResult.Err != nil || podResult.StatusCode != 200 {
+								return false
+							}
+						}
+					}
+					return true
+				}, 5, 150*time.Second)
+
+				Expect(success).To(BeTrue())
+
+				By("Disable Permissive, expect traffic to flow in SMI mode too")
+				Expect(setPermissiveMode(false)).To(BeNil())
+
+				success = Td.WaitForRepeatedSuccess(func() bool {
+					// Issue all calls, get results
+					results = Td.MultipleHTTPRequest(&requests)
+
+					// Care, depending on variables there could be a lot of results
+					Td.PrettyPrintHTTPResults(&results)
+
+					// Verify success conditions
+					for _, ns := range results {
+						for _, podResult := range ns {
+							if podResult.Err != nil || podResult.StatusCode != 200 {
+								return false
+							}
+						}
+					}
+					return true
+				}, 20, 150*time.Second)
+
+				Expect(success).To(BeTrue())
+			})
+		})
+	})
+
+func setPermissiveMode(b bool) error {
```

Comment:
Instead of creating this function, we could just keep the request success block as it's own function / context as mentioned in the above comment, and just set the permissive mode upon install: https://github.com/openservicemesh/osm/blob/main/tests/e2e/e2e_tcp_client_server_test.go#L40-L42